skbio/tree/_tree.py

```diff
@@ -3650,168 +4149,540 @@ def update_result():
         for node in self.postorder():
             if not node.children:
                 continue
-            # subtree with solved child wedges
+            # subtree with solved child edges
             # can possibly use np.zeros
-            starts, stops = [], []  # to calc ._start and ._stop for curr node
-            for child in node.children:
-                length = child.length
-                if length is None:
+            starts, stops = [], []
+            for child in (children := node.children):
```

Comment:
It looks like the use of the walrus is to leak the `children` alias for use outside of the for loop. Doesn't seem needed?